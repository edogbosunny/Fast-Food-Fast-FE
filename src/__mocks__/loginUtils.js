/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
