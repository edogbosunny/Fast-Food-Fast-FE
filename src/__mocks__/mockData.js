const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
