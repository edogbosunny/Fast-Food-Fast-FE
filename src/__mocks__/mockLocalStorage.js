let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
