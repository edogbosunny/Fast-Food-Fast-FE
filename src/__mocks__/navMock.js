/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
