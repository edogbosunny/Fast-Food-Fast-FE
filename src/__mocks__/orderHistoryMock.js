/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
