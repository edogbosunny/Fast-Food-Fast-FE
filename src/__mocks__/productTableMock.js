/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
