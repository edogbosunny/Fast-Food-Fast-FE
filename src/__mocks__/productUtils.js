/* eslint-disable import/prefer-default-export */
export const props = {
  addToCart: jest.fn(),
  auth: {
    isAuthenticated: true,
    loading: false,
    user: { id: 1, iat: 1549456186, exp: 1549542586 },
  },
  foodMenu: {
    foodMenu: {
      count: 3,
      data: [
        { meal_id: 1, price: '600', meal: 'Spaniish Noodles' },
        { meal_id: 2, price: '200', meal: 'DODO' },
        { meal_id: 3, price: '500', meal: 'Ewagoin' },
      ],
      message: 'Food menu has been retrieved succesfully',
    },
  },
  getAllMenu: jest.fn(),
  history: {
    push: jest.fn(),
  },
  alert: {
    show: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
  },
  loader: { loading: false },
  mealQuantity: undefined,
  staticContext: undefined,
};
