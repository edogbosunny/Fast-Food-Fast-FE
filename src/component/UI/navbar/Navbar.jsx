/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

// import '../../styles/ind.css';

class Navbar extends Component {
  render() {
    return (
      <nav className="navbar navbar-expand-sm navbar-dark nav-dark mb-4">
        <div className="container">
          <Link to='/' className='logo navbar-brand' />
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            <ul className="navbar-nav mr-auto nav-tabs">
              <li className="nav-item">
                <Link to='/' className="nav-link  m-auto">
                  {' '}
                  Home
                </Link>
              </li>
              <li className="nav-item ">
                <Link to='/about' className="nav-link m-auto">
                  {' '}
                  About
                </Link>
              </li>
            </ul>

            <ul className="navbar-nav ml-auto nav-tabs">
              <li className="nav-item">
                <a className="nav-link" href="register.html">
                  Sign Up
                </a>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="login.html">
                  Login
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

export default Navbar;
