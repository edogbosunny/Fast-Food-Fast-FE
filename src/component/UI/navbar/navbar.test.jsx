import React from 'react';
import { shallow } from 'enzyme';
import { Navbar, mapStateToProps } from './Navbar';
import { props, mockState } from '../../../__mocks__/navMock';

describe('Navbar Test Suit', () => {
  const wrapper = shallow(<Navbar {...props} />);
  it('should mount navbar', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(mockState);
    expect(state).toBeTruthy();
  });

  it('should test onlogout function', () => {
    const event = {
      target: { name: 'email', value: 'test@test.com' },
      preventDefault: () => {},
    };

    wrapper.instance().onLogoutClick(event);
  });
});
