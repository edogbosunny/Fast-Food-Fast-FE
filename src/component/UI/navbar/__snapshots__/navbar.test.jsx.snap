// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Test Suit should mount navbar 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-sm navbar-dark nav-dark mb-4"
  >
    <div
      className="container"
    >
      <Link
        className="logo navbar-brand"
        replace={false}
        to="/"
      />
      <button
        className="navbar-toggler"
        data-target="#mobile-nav"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="mobile-nav"
      >
        <ul
          className="navbar-nav mr-auto nav-tabs"
        >
          <li
            className="nav-item"
          >
            <Link
              className="nav-link  m-auto"
              replace={false}
              to="/"
            >
               
              Home
            </Link>
          </li>
          <li
            className="nav-item "
          >
            <Link
              className="nav-link m-auto"
              replace={false}
              to="/about"
            >
               
              About
            </Link>
          </li>
        </ul>
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              replace={false}
              to="/order-history"
            >
              Order History
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              onClick={[Function]}
              replace={false}
              to=""
            >
              Logout
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</Fragment>
`;
